go/cli/mcap/cmd/sort.go

```diff
@@ -0,0 +1,157 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/foxglove/mcap/go/mcap"
+	"github.com/spf13/cobra"
+)
+
+var (
+	sortOutputFile string
+)
+
+type errUnindexedFile struct {
+	err error
+}
+
+func (e errUnindexedFile) Error() string {
+	return e.err.Error()
+}
+
+func (e errUnindexedFile) Is(tgt error) bool {
+	_, ok := tgt.(errUnindexedFile)
+	return ok
+}
+
+func sortFile(w io.Writer, r io.ReadSeeker) error {
+	reader, err := mcap.NewReader(r)
+	if err != nil {
+		return fmt.Errorf("failed to create reader: %w", err)
+	}
+	writer, err := mcap.NewWriter(w, &mcap.WriterOptions{
+		Chunked:     true,
+		Compression: mcap.CompressionZSTD,
+		ChunkSize:   4 * 1024 * 1024,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create writer: %w", err)
+	}
+	info, err := reader.Info()
+	if err != nil {
+		return errUnindexedFile{err}
+	}
+
+	err = writer.WriteHeader(info.Header)
+	if err != nil {
+		return fmt.Errorf("failed to write header: %w", err)
+	}
+
+	// handle the attachments and metadata metadata first; physical location in
+	// the file is irrelevant but order is preserved.
+	for _, index := range info.AttachmentIndexes {
+		ar, err := reader.GetAttachment(index.Offset)
```

Comment:
I know that short variable names are common in this codebase but I cannot imagine what `ar` stands for.

Reply:
it stands for AttachmentReader, which is a type that unfortunately needs to be public and different from Attachment. I should perhaps rename this function to GetAttachmentReader - I guess we might as well rip the bandaid off.

Reply:
why not name the variable `attachmentReader` or `attReader`?

Reply:
sorry, missed this - was your proposal "attachmentReader := reader.GetAttachment()" or "attachmentReader := reader.GetAttachmentReader()"?

I don't really like the incongruity of the first, and the second to me doesn't make things clearer than "ar := reader.GetAttachmentReader()". Either way you need to figure out what an attachment reader is, if you don't know already.

Reply:
pushed up the change anyway

Reply:
yeah, but in the original code you also have to figure out what an `ar` is 😂